From 739d8fd834e9d6fa5056e1dc697ae2a2f26dd029 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Fri, 21 Mar 2014 21:01:50 +0800
Subject: [PATCH] Revert "clocksource: rockchip_timer: always
 setup_sched_clock"

This reverts commit 3a420a06f9c03ae65c91763af3ff8a1fb8391d6a.
---
 drivers/clocksource/rockchip_timer.c | 11 ++++++-----
 1 file changed, 6 insertions(+), 5 deletions(-)

diff --git a/drivers/clocksource/rockchip_timer.c b/drivers/clocksource/rockchip_timer.c
index 864b27c3fb6d..1f74db6a8210 100644
--- a/drivers/clocksource/rockchip_timer.c
+++ b/drivers/clocksource/rockchip_timer.c
@@ -321,12 +321,13 @@ static void __init rk_timer_init(struct device_node *np)
 			rk_delay_timer.read_current_timer = (unsigned long (*)(void))rockchip_read_sched_clock_up;
 		}
 		rk_timer_init_clocksource(np);
-		if (count_up)
-			setup_sched_clock(rockchip_read_sched_clock_up, 32, 24000000);
-		else
-			setup_sched_clock(rockchip_read_sched_clock, 32, 24000000);
-		if (!lpj_fine)
+		if (!lpj_fine) {
+			if (count_up)
+				setup_sched_clock(rockchip_read_sched_clock_up, 32, 24000000);
+			else
+				setup_sched_clock(rockchip_read_sched_clock, 32, 24000000);
 			register_current_timer_delay(&rk_delay_timer);
+		}
 	} else if (of_property_read_u32(np, "rockchip,broadcast", &val) == 0 && val) {
 		rk_timer_init_broadcast(np);
 	}
-- 
2.35.3

